python/cudf/cudf/core/frame.py

```diff
@@ -163,6 +163,16 @@ def _from_columns(
 
         return cls._from_data(data, index)
 
+    def _from_columns_like_self(
+        self,
+        columns: List[ColumnBase],
+        column_names: List[str],
+        index_names: Optional[List[str]] = None,
```

Comment:
Is the plan to eventually specialize this in `IndexedFrame`?

Reply:
I think we can, and we need to provide support to Index/MultiIndex too, in that case we would need create its counterpart in `_base_index`.

Reply:
We would need to move `_from_columns` too.